lib/dry/web/settings.rb

```diff
@@ -27,16 +26,39 @@ def self.check_schema_duplication(new_schema)
       end
       private_class_method :check_schema_duplication
 
+      def self.extract_environment_data(root, env)
+        env_file = find_env_file(root, env)
+        raise MissingEnvFile, "Try to load #{env} file, but no file found" unless env_file
```

Comment:
I'm not sure if we want to fail with an exception here. I think it'd be better to to just return a "null object" (i.e. empty hash) if no file was found.

The reason for this is that it's completely reasonable to expect no files in production, for example, where ENV vars are relied upon for everything.